tests/helpers_tests/test_condition.py

```diff
@@ -721,3 +800,504 @@ def test_audience_condition_deserializer_defaults(self):
+    mock_log.assert_called_once_with(
+      enums.AudienceEvaluationLogs.UNEXPECTED_TYPE.format(
+        json.dumps(expected_condition_log), type('48'), 'meters_travelled'
+      )
+    )
+
+  def test_less_than__user_value__unexpected_type(self):
+    log_level = 'debug'
+    lt_condition_list = [['meters_travelled', 48, 'custom_attribute', 'lt']]
+    user_attributes = {'meters_travelled': True}
+
+    with mock.patch('optimizely.logger.reset_logger', return_value=self.mock_client_logger):
+      evaluator = condition_helper.CustomAttributeConditionEvaluator(
+        lt_condition_list, user_attributes, self.mock_client_logger
+      )
+
+    expected_condition_log = {
+      "name": 'meters_travelled',
+      "value": 48,
+      "type": 'custom_attribute',
+      "match": 'lt'
+    }
+
+    self.assertIsNone(evaluator.evaluate(0))
+
+    mock_log = getattr(self.mock_client_logger, log_level)
+    mock_log.assert_called_once_with(
+      enums.AudienceEvaluationLogs.UNEXPECTED_TYPE.format(
+        json.dumps(expected_condition_log), type(True), 'meters_travelled'
+      )
+    )
+
+  def test_substring__user_value__unexpected_type(self):
+    log_level = 'debug'
+    substring_condition_list = [['headline_text', '12', 'custom_attribute', 'substring']]
+    user_attributes = {'headline_text': 1234}
+
+    with mock.patch('optimizely.logger.reset_logger', return_value=self.mock_client_logger):
+      evaluator = condition_helper.CustomAttributeConditionEvaluator(
+        substring_condition_list, user_attributes, self.mock_client_logger
+      )
+
+    expected_condition_log = {
+      "name": 'headline_text',
+      "value": '12',
+      "type": 'custom_attribute',
+      "match": 'substring'
+    }
+
+    self.assertIsNone(evaluator.evaluate(0))
+
+    mock_log = getattr(self.mock_client_logger, log_level)
+    mock_log.assert_called_once_with(
+      enums.AudienceEvaluationLogs.UNEXPECTED_TYPE.format(
+        json.dumps(expected_condition_log), type(1234), 'headline_text'
+      )
+    )
+
+  def test_exact__user_value__infinite(self):
+    log_level = 'debug'
+    exact_condition_list = [['favorite_constellation', 'Lacerta', 'custom_attribute', 'exact']]
+    user_attributes = {'favorite_constellation': int(2**53) + 1}
+
+    with mock.patch('optimizely.logger.reset_logger', return_value=self.mock_client_logger):
+      evaluator = condition_helper.CustomAttributeConditionEvaluator(
+        exact_condition_list, user_attributes, self.mock_client_logger
+      )
+
+    expected_condition_log = {
+      "name": 'favorite_constellation',
+      "value": 'Lacerta',
+      "type": 'custom_attribute',
+      "match": 'exact'
+    }
+
+    self.assertIsNone(evaluator.evaluate(0))
+
+    mock_log = getattr(self.mock_client_logger, log_level)
+    mock_log.assert_called_once_with(
+      enums.AudienceEvaluationLogs.INFINITE_ATTRIBUTE_VALUE.format(
+        json.dumps(expected_condition_log), int(2**53) + 1
```

Comment:
`int(2**53) + 1` doesn't (and shouldn't) actually appear in the logged string. We can clarify this by not using `format` in our assertions